src/probnum/prob/_random_variable.py

```diff
@@ -0,0 +1,904 @@
+        sample: Optional[Callable[[int], ValueType]] = None,
+        in_support: Optional[Callable[[ValueType], bool]] = None,
+        pdf: Optional[Callable[[ValueType], float]] = None,
+        logpdf: Optional[Callable[[ValueType], float]] = None,
+        cdf: Optional[Callable[[ValueType], float]] = None,
+        logcdf: Optional[Callable[[ValueType], float]] = None,
+        quantile: Optional[Callable[[float], ValueType]] = None,
+        mode: Optional[Callable[[], ValueType]] = None,
+        median: Optional[Callable[[], ValueType]] = None,
+        mean: Optional[Callable[[], ValueType]] = None,
+        cov: Optional[Callable[[], ValueType]] = None,
+        var: Optional[Callable[[], ValueType]] = None,
+        std: Optional[Callable[[], ValueType]] = None,
+        entropy: Optional[Callable[[], float]] = None,
+        properties: Optional[Dict[str, Any]] = None,
+    ):
+        """Create a new random variable."""
+        self._shape = RandomVariable._check_shape(shape)
+        self._dtype = dtype
+
+        self._random_state = scipy._lib._util.check_random_state(random_state)
+
+        # Probability distribution of the random variable
+        self._parameters = parameters.copy() if parameters is not None else {}
+
+        self.__sample = sample
+
+        self.__in_support = in_support
+        self.__pdf = pdf
+        self.__logpdf = logpdf
+        self.__cdf = cdf
+        self.__logcdf = logcdf
+        self.__quantile = quantile
+
+        # Properties of the random variable
+        if properties is None:
+            properties = {}
+
+        self.__mode = RandomVariable._get_property_fn("mode", mode, properties)
+        self.__median = RandomVariable._get_property_fn("median", median, properties)
+        self.__mean = RandomVariable._get_property_fn("mean", mean, properties)
+        self.__cov = RandomVariable._get_property_fn("cov", cov, properties)
+        self.__var = RandomVariable._get_property_fn("var", var, properties)
+        self.__std = RandomVariable._get_property_fn("std", std, properties)
+        self.__entropy = RandomVariable._get_property_fn("entropy", entropy, properties)
+
+        # Further processing
+        if self._shape is None or self._dtype is None:
+            inferred_shape, inferred_dtype = self._infer_shape_and_dtype()
+
+            if self._shape is None:
+                self._shape = inferred_shape
+
+            if self._dtype is None:
+                self._dtype = inferred_dtype
+
+    @staticmethod
+    def _check_shape(
+        shape: Optional[Union[int, Tuple[int, ...]]]
+    ) -> Optional[Tuple[int, ...]]:
+        if shape is None:
+            return None
+        elif isinstance(shape, tuple) and all(
+            isinstance(entry, int) for entry in shape
+        ):
+            return shape
+        elif isinstance(shape, int):
+            return (shape,)
+        else:
+            raise TypeError("The given shape is not an int or a tuple of ints.")
```

Comment:
I'd suggest to return the given shape in the Error message to make debugging easier, also in the light of #67 , i.e.
```python
raise TypeError(f"The given shape {str(shape)} is not an int or a tuple of ints.")
```

Reply:
I'm not here to actually review but since I saw it while passing: Calling `str` explicitly should not be necessary for this